arbnode/node.go

```diff
@@ -520,42 +527,68 @@ func createNodeImpl(
 		return nil, err
 	}
 
-	var daWriter das.DataAvailabilityServiceWriter
-	var daReader das.DataAvailabilityServiceReader
-	var dasLifecycleManager *das.LifecycleManager
-	var dasKeysetFetcher *das.KeysetFetcher
-	if config.DataAvailability.Enable {
-		if config.BatchPoster.Enable {
-			daWriter, daReader, dasKeysetFetcher, dasLifecycleManager, err = das.CreateBatchPosterDAS(ctx, &config.DataAvailability, dataSigner, l1client, deployInfo.SequencerInbox)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			daReader, dasKeysetFetcher, dasLifecycleManager, err = das.CreateDAReaderForNode(ctx, &config.DataAvailability, l1Reader, &deployInfo.SequencerInbox)
-			if err != nil {
-				return nil, err
-			}
+	if config.DAProvider.Enable && config.DataAvailability.Enable {
+		return nil, errors.New("da-provider and data-availability cannot be enabled together")
+	}
+
+	var daClient *daclient.Client
+	var withDAWriter bool
+	var dasServerCloseFn func()
+	if config.DAProvider.Enable {
+		daClient, err = daclient.NewClient(ctx, func() *rpcclient.ClientConfig { return &config.DAProvider.RPC })
+		if err != nil {
+			return nil, err
 		}
+		// Only allow dawriter if batchposter is enabled
+		withDAWriter = config.DAProvider.WithWriter && config.BatchPoster.Enable
+	} else if config.DataAvailability.Enable {
+		jwtPath := path.Join(filepath.Dir(stack.InstanceDir()), "dasserver-jwtsecret")
+		if err := genericconf.TryCreatingJWTSecret(jwtPath); err != nil {
+			return nil, fmt.Errorf("error writing ephemeral jwtsecret of dasserver to file: %w", err)
+		}
+		log.Info("Generated ephemeral JWT secret for dasserver", "jwtPath", jwtPath)
+		// JWTSecret is no longer needed, cleanup when returning
+		defer func() {
+			if err := os.Remove(jwtPath); err != nil {
+				log.Error("error deleting generated ephemeral JWT secret of dasserver", "jwtPath", jwtPath)
+			}
+		}()
 
-		daReader = das.NewReaderTimeoutWrapper(daReader, config.DataAvailability.RequestTimeout)
+		serverConfig := dasserver.DefaultServerConfig
+		serverConfig.Port = 0 // Initializes server at a random available port
+		serverConfig.DataAvailability = config.DataAvailability
+		serverConfig.EnableDAWriter = config.BatchPoster.Enable
+		serverConfig.JWTSecret = jwtPath
+		withDAWriter = config.BatchPoster.Enable
+		dasServer, closeFn, err := dasserver.NewServer(ctx, &serverConfig, dataSigner, l1client, l1Reader, deployInfo.SequencerInbox)
+		if err != nil {
+			return nil, err
+		}
 
```

Comment:
If this is being initialized in the same process what's the point of communicating over TCP at all?

Reply:
I think the reasoning here is we would like to demonstrate that the current design and implementation works.

Redirecting to @PlasmaPower for confirmation.